src/MIEApi.js

```diff
@@ -0,0 +1,138 @@
+import axios from 'axios';
+import logger from './logger.js'; 
+import { endpoints } from './apiConfig.js';
+
+class MIEApi {
+  static sessionCache = new Map();
+
+  constructor({ baseUrl, connectToken, userId, ip }) {
+    this.baseUrl = baseUrl;
+    this.connectToken = connectToken;
+    this.userId = userId;
+    this.ip = ip;
+    this.isRefreshingSession = false;
+  }
+
+  static getSessionKey(baseUrl, userId) {
+    return `${baseUrl}_${userId}`;
+  }
+
+  async refreshSession() {
+    if (this.isRefreshingSession) return;
+    this.isRefreshingSession = true;
+
+    try {
+      const refreshUrl = `${this.baseUrl}?f=layoutnouser&name=BlueHive_Refresh_Session&user_id=${encodeURIComponent(this.userId)}&connectToken=${encodeURIComponent(this.connectToken)}&ip_address=${encodeURIComponent(this.ip)}&raw&json`;
+
+      logger.info(`Refreshing connection to: ${refreshUrl}, 'for userId:', ${this.userId}`);
+
+      const response = await axios.get(refreshUrl, {
+        headers: {
+          'User-Agent': 'BlueHive AI (Refresh Connection)',
```

Comment:
Dup?

Reply:
Also, please do not use this user agent. You should call it something like mieapi and possibly include the version string for the NPM version.

Reply:
I changed it to mieapi and included the npm version

![image](https://github.com/user-attachments/assets/6bd199b0-9b1d-4e4b-87a5-91b05e566e34)